R/select.R

```diff
@@ -101,3 +101,46 @@ check_data_types <- function(data, register, call = rlang::caller_env()) {
   }
   data
 }
+#' Adapt LPR_A data to LPR3 schema
+#'
+#' Converts variable names and types from the "LPR_A" format (`lpr_a_kontakt`,
+#' `lpr_a_diagnose`) to match the LPR3 schema expected by [prepare_lpr3()].
+#'
+#' E.g. `kont_starttidspunkt` is renamed to `dato_start` and converted to a
+#' date type, whereas `diag_kode` is simply renamed to `diagnosekode`.
+#'
+#' @param lpr_a_kontakt A duckplyr tibble containing contact data in
+#' LPR A format (typically data from 2022 onward).
+#' @param lpr_a_diagnose A duckplyr tibble containing diagnosis data in
+#' LPR A format.
+#'
+#' @return A list containing two tables, `kontakter` and `diagnoser`,
+#' with column names and types adapted to match the LPR3 format.
+#' @keywords internal
+adapt_lpr_a_to_lpr3 <- function(lpr_a_kontakt, lpr_a_diagnose) {
```

Comment:
This really should belong in `prepare_lpr.R` file, as part of a `prepare_lpra()` function.